src/main/java/gameframework/game/GameUniverseDefaultImpl.java

```diff
@@ -21,34 +21,57 @@
 	@Override
 	public synchronized void addGameEntity(GameEntity gameEntity) {
 		gameEntities.add(gameEntity);
-		if (gameEntity instanceof Overlappable) {
-			getOverlapProcessor().addOverlappable((Overlappable) gameEntity);
-		}
-		if (gameEntity instanceof MoveBlocker) {
-			getMoveBlockerChecker().addMoveBlocker((MoveBlocker) gameEntity);
-		}
+		addMotionEntity(gameEntity);
```

Comment:
This will never work in practice as I explained during the COO lecture. At compilation time, `gameEntity` has the `GameEntity` static type and thus the empty `addMotionEntity(GameEntity)` will always be called instead of the more specialized methods.